packages/bunyan/src/index.ts

```diff
@@ -0,0 +1,98 @@
+import { Axiom } from '@axiomhq/js';
+
+export enum AxiomEventLevel {
+  Trace = 'trace',
+  Debug = 'debug',
+  Info = 'info',
+  Warn = 'warn',
+  Error = 'error',
+  Fatal = 'fatal',
+  Silent = 'silent',
+}
+
+enum RecordLevel {
+  trace = 10,
+  debug = 20,
+  info = 30,
+  warn = 40,
+  error = 50,
+  fatal = 60,
+}
+
+const noop = () => {};
+
+export class AxiomStream<T extends { msg: string; level: RecordLevel; time: Date }> {
+  private axiom: Axiom;
+  dataset: string;
+  token: string;
+  orgId?: string;
+  onError: (err: unknown) => void;
+
+  constructor(
+    opt: Omit<AxiomStream<T>, 'write' | 'onError' | 'close'> & {
+      onError?: AxiomStream<T>['onError'];
+    },
+  ) {
+    if (!opt.dataset) {
+      throw new Error('dataset requried');
+    }
+
+    if (!opt.token) {
+      throw new Error('token requried');
+    }
+
+    this.orgId = opt.orgId;
+    this.token = opt.token;
+    this.onError = opt.onError || noop;
+    this.dataset = opt.dataset;
+
+    this.axiom = new Axiom({
+      orgId: this?.orgId,
+      token: this.token,
+    });
+  }
+
+  async write(record: Object) {
+    try {
+      const parsedRecord = typeof record === 'string' ? (JSON.parse(record) as T) : (record as T);
+      const { time, level, msg, ...rest } = parsedRecord;
+      const event = {
+        _time: time,
+        message: msg,
+        level: this.getLogLevel(level),
+        ...rest,
+      };
+
+      await this.axiom.ingest(this.dataset!, event);
+    } catch (err) {
+      this.onError(err);
+    }
+  }
+
+  private getLogLevel(level: number) {
+    if (level <= 10) {
+      return AxiomEventLevel.Trace;
+    }
+    if (level <= 20) {
+      return AxiomEventLevel.Debug;
+    }
+    if (level <= 30) {
+      return AxiomEventLevel.Info;
+    }
+    if (level <= 40) {
+      return AxiomEventLevel.Warn;
+    }
+    if (level <= 50) {
+      return AxiomEventLevel.Error;
+    }
+    if (level <= 60) {
+      return AxiomEventLevel.Fatal;
+    }
+
+    return AxiomEventLevel.Silent;
+  }
+
+  async close() {
```

Comment:
Can we call this `flush`?

Reply:
how does this work, is it something that bunyan calls , or does the user has to call it? couple of comments would be great here.

Reply:
The Bunyan logger doesn't automatically call close or flush out of the box; it needs to be called manually.

Reply:
can you update the function to `flush`? then we can proceed to merge